lib/utils.pm

```diff
@@ -533,7 +533,13 @@ sub zypper_call {
 
     unless (grep { $_ == $ret } @$allow_exit_codes) {
         upload_logs('/var/log/zypper.log');
-        die "'zypper -n $command' failed with code $ret";
+        my $msg = "'zypper -n $command' failed with code $ret";
+        if ($ret == 104) {
+            $msg .= " (ZYPPER_EXIT_INF_CAP_NOT_FOUND)\n\nRelated zypper logs:\n";
+            script_run('tac /var/log/zypper.log | grep -F -m1 -B10000 "Hi, me zypper" | tac | grep \'\(SolverRequester.cc\|THROW\|CAUGHT\)\' > /tmp/z104.txt');
```

Comment:
Why not search just for: `SolverRequester.cc.* cap:machinery not found`? however I'm ok with the result, as it captures properly the actual problem and possibly points to the root cause

```suggestion
            script_run('tac /var/log/zypper.log | grep -F -m1 -B10000 "Hi, me zypper" | tac | grep \'SolverRequester.cc .* cap:.* not found\' > /tmp/z104.txt');
```

Reply:
I would prefer the other way as it provides some additional info.
Also your regex doesn't seem to work: http://kazhua.qa.suse.de/tests/75#step/machinery/38